### PR TITLE
feat: Preview App header revamp

### LINKED ISSUE #8355: Viewer app header revamp
The viewer's header needs a revamp.

### PR DESCRIPTION
**Pull request by nakulnagargade**
Resolves #8355.

### CODE REVIEW

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Hey @nakulnagargade , there is some inconsistency w.r.t to dark mode in this area:

image


Other than that, looks like there are a few undeleted `console.log`s.

I felt there is some kind of alignment issue at the same place, but don't have design to compare it with, please share design as well

**Comment by kavinvenkatachalam** [comment]
@nakulnagargade Please resolve the conflicts

**Comment by nakulnagargade** [comment]
@kavinvenkatachalam Merge conflict resolved.

File: frontend/src/Editor/Header/HeaderActions.jsx [hunk 1]
```diff
@@ -15,94 +23,102 @@ function HeaderActions({ handleUndo, canUndo, handleRedo, canRedo }) {
...
+                name="computer"
+                width="14"
+                fill={currentLayout === 'desktop' ? 'var(--slate12)' : 'var(--slate8)'}
+              />
+            </button>
+            <button
+              className={cx('btn border-0 p-1', {
+                'bg-transparent': currentLayout !== 'mobile',
+                'bg-white': currentLayout === 'mobile',
+                'opacity-100': currentLayout === 'mobile',
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Header/HeaderActions.jsx:60]:
```suggestion
                'bg-white opacity-100': currentLayout === 'mobile',
```

File: frontend/src/Editor/Header/HeaderActions.jsx [hunk 2]
```diff
@@ -15,94 +23,102 @@ function HeaderActions({ handleUndo, canUndo, handleRedo, canRedo }) {
...
+            className="d-flex align-items-center p-1 current-layout"
+            style={{ height: 28, background: darkMode ? '#202425' : '#F1F3F5', borderRadius: 6 }}
+            role="tablist"
+            aria-orientation="horizontal"
+          >
+            <button
+              className={cx('btn border-0 p-1', {
+                'bg-transparent': currentLayout !== 'desktop',
+                'bg-white': currentLayout === 'desktop',
+                'opacity-100': currentLayout === 'desktop',
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Header/HeaderActions.jsx:39]:
```suggestion
                'bg-white opacity-100': currentLayout === 'desktop',
```

File: frontend/src/Editor/Header/index.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ import AppLogo from '@/_components/AppLogo';
 import EditAppName from './EditAppName';
 import HeaderActions from './HeaderActions';
 import RealtimeAvatars from '../RealtimeAvatars';
-import { AppVersionsManager } from '../AppVersionsManager/List';
+import { AppVersionsManager } from '../AppVersionsManager/AppVersionsManager';
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Header/index.js:7]: Can we import it using `@`?

File: frontend/src/Editor/Viewer/ViewerSidebarNavigation.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+export const ViewerSidebarNavigation = ({ isMobileDevice, pages, currentPageId, switchPage, darkMode }) => {
+  if (isMobileDevice) {
+    return null;
+  }
+  return (
+    <div
+      className={`navigation-area`}
+      style={{
+        width: 200,
+        // backgroundColor: canvasBackgroundColor,
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Viewer/ViewerSidebarNavigation.jsx:15]:
```suggestion
```

File: frontend/src/Editor/Viewer/MobileHeader.jsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+      <>
+        {showViewerNavigation ? (
+          <MobileNavigationMenu
+            pages={pages}
+            currentPageId={currentPageId}
+            switchPage={switchPage}
+            darkMode={darkMode}
+            changeDarkMode={changeDarkMode}
+            showHeader={showHeader}
+          />
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Viewer/MobileHeader.jsx:64]: Can we move the repeated codes like `PreviewSettings`, `MobileNavigationMenu`, `span` etc to a function and call the function?

File: frontend/src/Editor/Viewer/PreviewSettings.jsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
+import React, { useState } from 'react';
+import Icon from '@/_ui/Icon/solidIcons/index';
+import { OverlayTrigger } from 'react-bootstrap';
+import { useAppVersionStore } from '@/_stores/appVersionStore';
+import { useAppInfo } from '@/_stores/appDataStore';
+import { AppVersionsManager } from '../AppVersionsManager/AppVersionsManager';
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Viewer/PreviewSettings.jsx:6]: If possible, can we use `@` to import the components?

File: frontend/src/Editor/Viewer/MobileNavigationMenu.jsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import _ from 'lodash';
+// eslint-disable-next-line import/no-unresolved
+import { slide as MobileMenu } from 'react-burger-menu';
+import { DarkModeToggle } from '@/_components/DarkModeToggle';
+import Header from './Header';
+import Cross from '@/_ui/Icon/solidIcons/Cross';
+
+const MobileNavigationMenu = ({ pages, switchPage, currentPageId, darkMode, changeDarkMode }) => {
+  const [hamburgerMenuOpen, setHamburgerMenuOpen] = React.useState(false);
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Viewer/MobileNavigationMenu.jsx:10]:
```suggestion
  const [hamburgerMenuOpen, setHamburgerMenuOpen]
 = useState(false);
```

We can follow importing the `useState` and use it

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@nakulnagargade Requesting few changes,

- In mobile layout, on hovering the close button the cursor needs to be changed to pointer (consult with Jaseem on this)
- The preview button's ZIndex should be lesser than the Modal. Attaching the screenshot
- The pages sidebar should be fixed when we scroll the webpage. Attaching the screenshot
- When the menu is disabled, the dark mode button on mobile seems misplaced. Attaching the screenshot
- Do we need the preview button on the header while launching the app?
- If we need the preview header while launching the app, then it doesn't suit the darkMode. Attaching the screenshot
Screenshot 2024-01-25 at 10 46 34 AM
Screenshot 2024-01-25 at 10 51 35 AM

Screenshot 2024-01-25 at 10 56 10 AM
Screenshot 2024-01-25 at 11 00 21 AM
Screenshot 2024-01-25 at 11 47 46 AM

**Comment by nakulnagargade** [comment]
@kavinvenkatachalam Comments are resolved.

File: frontend/src/Editor/Header/HeaderActions.jsx (hunk 1 above)
Comment (nakulnagargade) [reply frontend/src/Editor/Header/HeaderActions.jsx]: Done

File: frontend/src/Editor/Header/HeaderActions.jsx (hunk 2 above)
Comment (nakulnagargade) [reply frontend/src/Editor/Header/HeaderActions.jsx]: Done

File: frontend/src/Editor/Header/index.js (hunk 3 above)
Comment (nakulnagargade) [reply frontend/src/Editor/Header/index.js]: Done

File: frontend/src/Editor/Viewer/ViewerSidebarNavigation.jsx (hunk 4 above)
Comment (nakulnagargade) [reply frontend/src/Editor/Viewer/ViewerSidebarNavigation.jsx]: Done

File: frontend/src/Editor/Viewer/MobileHeader.jsx (hunk 5 above)
Comment (nakulnagargade) [reply frontend/src/Editor/Viewer/MobileHeader.jsx]: Done

File: frontend/src/Editor/Viewer/PreviewSettings.jsx (hunk 6 above)
Comment (nakulnagargade) [reply frontend/src/Editor/Viewer/PreviewSettings.jsx]: Done

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@nakulnagargade When I hover the dark mode button, the scrollbar seems to be visible. Attaching a GIF, rest everything looks good

Jan-30-2024 15-32-31

**Comment by nakulnagargade** [comment]
> @nakulnagargade When I hover the dark mode button, the scrollbar seems to be visible. Attaching a GIF, rest everything looks good
> 
> Jan-30-2024 15-32-31 [ Jan-30-2024 15-32-31 ](https://private-user-images.githubusercontent.com/50441969/300764091-605ea342-fff7-4261-aa01-85734733d2e5.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2NzYyMjksIm5iZiI6MTcwNjY3NTkyOSwicGF0aCI6Ii81MDQ0MTk2OS8zMDA3NjQwOTEtNjA1ZWEzNDItZmZmNy00MjYxLWFhMDEtODU3MzQ3MzNkMmU1LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDA0Mzg0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5ODM1MzRmMWRlMWU2YjA1ODEyMDgxN2VhNzZmY2FlZDI4NjUzNmJmZTRmOTgxM2FlYWI3NjI5NDAxZTgwNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AU-YZVCgwDp8ETT7iFiykMHkPb3MmcO1k2ROwzY7jbA) [ ](https://private-user-images.githubusercontent.com/50441969/300764091-605ea342-fff7-4261-aa01-85734733d2e5.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2NzYyMjksIm5iZiI6MTcwNjY3NTkyOSwicGF0aCI6Ii81MDQ0MTk2OS8zMDA3NjQwOTEtNjA1ZWEzNDItZmZmNy00MjYxLWFhMDEtODU3MzQ3MzNkMmU1LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDA0Mzg0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5ODM1MzRmMWRlMWU2YjA1ODEyMDgxN2VhNzZmY2FlZDI4NjUzNmJmZTRmOTgxM2FlYWI3NjI5NDAxZTgwNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AU-YZVCgwDp8ETT7iFiykMHkPb3MmcO1k2ROwzY7jbA)

@kavinvenkatachalam This is resolved.

**Review by kavinvenkatachalam — APPROVED** [review]

**Pull request merged**